staging/src/k8s.io/apiserver/pkg/server/options/recommended.go

```diff
@@ -55,26 +60,33 @@ func (o *RecommendedOptions) AddFlags(fs *pflag.FlagSet) {
 	o.Features.AddFlags(fs)
 }
 
-func (o *RecommendedOptions) ApplyTo(config *server.Config) error {
-	if err := o.Etcd.ApplyTo(config); err != nil {
+func (o *RecommendedOptions) ApplyTo(config *server.RecommendedConfig) error {
+	if err := o.Etcd.ApplyTo(&config.Config); err != nil {
 		return err
 	}
-	if err := o.SecureServing.ApplyTo(config); err != nil {
+	if err := o.SecureServing.ApplyTo(&config.Config); err != nil {
 		return err
 	}
-	if err := o.Authentication.ApplyTo(config); err != nil {
+	if err := o.Authentication.ApplyTo(&config.Config); err != nil {
 		return err
 	}
-	if err := o.Authorization.ApplyTo(config); err != nil {
+	if err := o.Authorization.ApplyTo(&config.Config); err != nil {
 		return err
 	}
-	if err := o.Audit.ApplyTo(config); err != nil {
+	if err := o.Audit.ApplyTo(&config.Config); err != nil {
 		return err
 	}
-	if err := o.Features.ApplyTo(config); err != nil {
+	if err := o.Features.ApplyTo(&config.Config); err != nil {
 		return err
 	}
 
+	// do convenience work for RecommendedOptions users
+	clientgoExternalClient, err := clientgoclientset.NewForConfig(config.LoopbackClientConfig)
```

Comment:
This is what @p0lyn0mial discovered was odd.  This should be an in-cluster-config, not a loopback.

Reply:
right, I remember having seen that in the old code. will fix.